src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts

```diff
@@ -0,0 +1,149 @@
+import { html, LitElement, TemplateResult, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import { stopPropagation } from "../../../../../common/dom/stop_propagation";
+import { HassDialog } from "../../../../../dialogs/make-dialog-manager";
+import { changeZHANetworkChannel } from "../../../../../data/zha";
+import { showAlertDialog } from "../../../../../dialogs/generic/show-dialog-box";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import { HomeAssistant } from "../../../../../types";
+import "../../../../../components/buttons/ha-progress-button";
+import "../../../../../components/ha-button";
+import "../../../../../components/ha-select";
+import "@material/mwc-list/mwc-list-item";
+import { ZHAMigrateChannelDialogParams } from "./show-dialog-zha-migrate-channel";
+
+const VALID_CHANNELS = [
+  "auto",
+  11,
+  12,
+  13,
+  14,
+  15,
+  16,
+  17,
+  18,
+  19,
+  20,
+  21,
+  22,
+  23,
+  24,
+  25,
+];
+
+@customElement("dialog-zha-migrate-channel")
+class DialogZHAMigrateChannel extends LitElement implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _migrationInProgress = false;
+
+  @state() private _params?: ZHAMigrateChannelDialogParams;
+
+  private _newChannel: "auto" | number = "auto";
```

Comment:
`_newChannel` should be initialized with `params.currentChannel` I think, and not with `auto`?